src/objectExplorer/objectExplorerService.ts

```diff
@@ -179,34 +182,34 @@ export class ObjectExplorerService {
 
                     // node expansion
                     let promise = new Deferred<TreeNodeInfo[]>();
-                    this.expandNode(element, element.sessionId, promise);
-                    return promise.then((children) => {
-                        if (children) {
-                            // clean expand session promise
-                            for (const key of this._expandParamsToPromiseMap.keys()) {
-                                if (key.sessionId === element.sessionId &&
-                                    key.nodePath === element.nodePath) {
-                                    this._expandParamsToPromiseMap.delete(key);
-                                }
+                    await this.expandNode(element, element.sessionId, promise);
+                    let children = await promise;
+                    if (children) {
+                        // clean expand session promise
```

Comment:
Why are you only removing the promise if children is undefined? The promise is done at this point - there's nothing else awaiting on it - so there's no point keeping it around even if no children were returned